rq/worker.py

```diff
@@ -225,6 +218,16 @@ def __init__(self, queues, name=None, default_result_ttl=DEFAULT_RESULT_TTL,
 
         self.disable_default_exception_handler = disable_default_exception_handler
 
+        if prepare_for_work:
+            self.hostname = socket.gethostname()
+            self.pid = os.getpid()
+            connection.client_setname(self.name)
```

Comment:
What does this actually do? I'm not sure why we're making changes to Redis connection.

Reply:
It sets the client name of the redis connection. It is used for getting the current connection to get the ip from.

Try `CLIENT LIST` in the redis cli.